common/ledger/util/leveldbhelper/leveldb_helper_test.go

```diff
@@ -157,6 +157,33 @@ func TestFileLock(t *testing.T) {
 	require.NoError(t, os.RemoveAll(fileLockPath))
 }
 
+func TestFileLockLockUnlockLock(t *testing.T) {
+	// create an open lock
+	lockPath := testDBPath + "/fileLock"
+	lock := NewFileLock(lockPath)
+	require.Nil(t, lock.db)
+	require.Equal(t, lock.filePath, lockPath)
+	require.False(t, lock.IsLocked())
+
+	defer lock.Unlock()
+	defer os.RemoveAll(lockPath)
```

Comment:
I think that your intent was the following. The `defers` are executed in reverse order.
```suggestion
	defer func(){
	     lock.Unlock()
	     os.RemoveAll(lockPath)
	}()
```